flatfs/flatfs.go

```diff
@@ -314,52 +315,33 @@ func (fs *Datastore) Query(q query.Query) (query.Results, error) {
 		return nil, errors.New("flatfs only supports listing all keys in random order")
 	}
 
-	// TODO this dumb implementation gathers all keys into a single slice.
-	root, err := os.Open(fs.path)
-	if err != nil {
-		return nil, err
-	}
-	defer root.Close()
+	reschan := make(chan query.Result)
+	go func() {
+		defer close(reschan)
+		err := filepath.Walk(fs.path, func(path string, info os.FileInfo, err error) error {
 
-	var res []query.Entry
-	prefixes, err := root.Readdir(0)
-	if err != nil {
-		return nil, err
-	}
-	for _, fi := range prefixes {
-		var err error
-		res, err = fs.enumerateKeys(fi, res)
-		if err != nil {
-			return nil, err
-		}
-	}
-	return query.ResultsWithEntries(q, res), nil
-}
+			if !info.Mode().IsRegular() || info.Name()[0] == '.' {
```

Comment:
Oh name[0], we're good.
On Sun, Dec 13, 2015 at 22:10 Jeromy Johnson <notifications@github.com>
wrote:

> In flatfs/flatfs.go
> <https://github.com/jbenet/go-datastore/pull/34#discussion_r47458423>:
>
> >
> > -	var res []query.Entry
> > -	prefixes, err := root.Readdir(0)
> > -	if err != nil {
> > -		return nil, err
> > -	}
> > -	for _, fi := range prefixes {
> > -		var err error
> > -		res, err = fs.enumerateKeys(fi, res)
> > -		if err != nil {
> > -			return nil, err
> > -		}
> > -	}
> > -	return query.ResultsWithEntries(q, res), nil
> > -}
> > +			if !info.Mode().IsRegular() || info.Name()[0] == '.' {
>
> I think the intent of this check was to ignore the random dotfiles that
> os'es (OSX) will randomly decide to put in folders. Other than that, i'm
> not certain.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jbenet/go-datastore/pull/34/files#r47458423>.
>